confirm.c

```diff
@@ -763,9 +855,35 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
+confirm_query_create6_tcp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
+		uint16_t src_port, uint16_t dst_port, confirm_query_cb cb)
+{
+	struct confirm_query *query = confirm_query_create6(dst, ttl, traffic_class,
+			flow_label, flowid, cb);
+
+	query->src_port = src_port;
+	query->dst_port = dst_port;
+	query->type = QUERY_TYPE_TCP;
+	return query;
+}
+
+struct confirm_query *
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
+{
+	struct confirm_query *query = confirm_query_create6(dst, ttl, traffic_class,
+			flow_label, flowid, cb);
+
+	query->icmpid = icmpid;
+	query->type = QUERY_TYPE_ICMP;
+	return query;
+}
+
+struct confirm_query *
 confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
+		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
```

Comment:
It might be better to have two... a confirm_query_create_tcp6 and _icmp6. I
understand this is not super clean, but it is the way netlib does it, so at
least people will need to get familiar with one interface only.

On Sun, Apr 17, 2016 at 2:48 PM, Rafael Almeida <notifications@github.com>
wrote:

> In confirm.c
> <https://github.com/cunha/netlib/pull/12#discussion_r59989093>:
>
> >  confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
> > -		uint8_t traffic_class, uint32_t flow_label,
> > -		uint16_t icmpid, uint8_t flowid,
> > +		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
>
> I tried to put the common things to tcp and icmp in this function. I think
> it should be a static function, right?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/cunha/netlib/pull/12/files/fa60c8b6ae6f38c64770ed45ebe80e4f1a356d1b#r59989093>
>